diff --git "a/shim.c" "b/shim.c"
index c5cfbb8..dc4846f 100644
--- "a/shim.c"
+++ "b/shim.c"
@@ -1413,7 +1413,7 @@ EFI_STATUS set_second_stage (EFI_HANDLE image_handle)
 
 	/* Sanity check since we make several assumptions about the length */
 	if (li->LoadOptionsSize % 2 != 0)
-		return EFI_INVALID_PARAMETER;
+		return EFI_SUCCESS;
 
 	/* So, load options are a giant pain in the ass.  If we're invoked
 	 * from the EFI shell, we get something like this:
